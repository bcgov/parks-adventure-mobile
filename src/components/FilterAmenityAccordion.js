import React from 'react'
import PropTypes from 'prop-types'
import { List } from 'react-native-paper'
import FilterAmenityListItem from '../components/FilterAmenityListItem'

const FilterAmenityAccordion = ({
  amenities,
  onSelect,
  expanded,
  onPress,
  amenityType = 'amenities',
}) => (
  <>
    <FilterAmenityListItem
      key={amenities[0].id}
      name={amenities[0].name}
      selected={amenities[0].selected}
      onSelect={() => onSelect(amenities[0].id)}
    />
    <FilterAmenityListItem
      key={amenities[1].id}
      name={amenities[1].name}
      selected={amenities[1].selected}
      onSelect={() => onSelect(amenities[1].id)}
    />
    <FilterAmenityListItem
      key={amenities[2].id}
      name={amenities[2].name}
      selected={amenities[2].selected}
      onSelect={() => onSelect(amenities[2].id)}
    />
    <List.Accordion
      title={`${!expanded ? 'See' : 'Collapse'} all ${amenityType}`}
      titleStyle={{ textDecorationLine: 'underline' }}
      expanded={expanded}
      onPress={onPress}>
      {amenities &&
        amenities
          .slice(3, amenities.length)
          .map((amenity) => (
            <FilterAmenityListItem
              key={amenity.id}
              name={amenity.name}
              selected={amenity.selected}
              onSelect={() => onSelect(amenity.id)}
            />
          ))}
    </List.Accordion>
  </>
)

FilterAmenityAccordion.propTypes = {
  amenities: PropTypes.array.isRequired,
  onSelect: PropTypes.func.isRequired,
  expanded: PropTypes.bool.isRequired,
  onPress: PropTypes.func.isRequired,
  amenityType: PropTypes.string,
}

export default FilterAmenityAccordion
