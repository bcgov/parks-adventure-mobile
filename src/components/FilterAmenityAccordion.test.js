import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import renderSnapshot from 'react-test-renderer'
import FilterAmenityAccordion from './FilterAmenityAccordion'

const amenities = [
  { id: '1', name: 'Hiking', selected: false },
  { id: '2', name: 'Fishing', selected: false },
  { id: '3', name: 'Climbing', selected: false },
  { id: '4', name: 'Wilderness Camping', selected: false },
]

function Wrapper() {
  const [expanded, setExpanded] = React.useState(false)

  return (
    <FilterAmenityAccordion
      amenities={amenities}
      onSelect={() => {}}
      expanded={expanded}
      onPress={() => setExpanded(true)}
    />
  )
}

test('Should expand and change text when onPress fired', () => {
  const { getByText } = render(<Wrapper />)

  const openButton = getByText('See all amenities')
  expect(openButton).toBeDefined()
  fireEvent.press(openButton)

  const closeButton = getByText('Collapse all amenities')
  expect(closeButton).toBeDefined()
})

test('matches snapshot', () => {
  const tree = renderSnapshot
    .create(
      <FilterAmenityAccordion
        amenities={amenities}
        onSelect={() => {}}
        expanded={true}
        onPress={() => {}}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
