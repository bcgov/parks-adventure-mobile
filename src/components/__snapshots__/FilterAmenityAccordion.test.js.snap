// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 8,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 0,
        },
      ]
    }
  >
    <SvgMock
      height={36}
      width={36}
    />
    <Text
      selected={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.53)",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 18,
              "lineHeight": 23,
              "marginLeft": 30,
            },
          ],
        ]
      }
    >
      Hiking
    </Text>
    <View
      accessibilityLabel="select"
      selected={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "black",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Icon
        color="white"
        icon="check"
        onPress={[Function]}
        size={26}
      />
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 8,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 0,
        },
      ]
    }
  >
    <SvgMock
      height={36}
      width={36}
    />
    <Text
      selected={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.53)",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 18,
              "lineHeight": 23,
              "marginLeft": 30,
            },
          ],
        ]
      }
    >
      Fishing
    </Text>
    <View
      accessibilityLabel="select"
      selected={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "black",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Icon
        color="white"
        icon="check"
        onPress={[Function]}
        size={26}
      />
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 8,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 0,
        },
      ]
    }
  >
    <SvgMock
      height={36}
      width={36}
    />
    <Text
      selected={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.53)",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 18,
              "lineHeight": 23,
              "marginLeft": 30,
            },
          ],
        ]
      }
    >
      Climbing
    </Text>
    <View
      accessibilityLabel="select"
      selected={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "black",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Icon
        color="white"
        icon="check"
        onPress={[Function]}
        size={26}
      />
    </View>
  </View>,
  <View>
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Array [
            Object {
              "padding": 8,
            },
            undefined,
          ],
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              Object {
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#6200ee",
                  },
                  Object {
                    "textDecorationLine": "underline",
                  },
                ],
              ]
            }
          >
            Collapse all amenities
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              undefined,
            ]
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Array [
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "material-community",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ï…ƒ
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 8,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 0,
          },
        ]
      }
    >
      <Text
        selected={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.53)",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontSize": 18,
                "lineHeight": 23,
                "marginLeft": 30,
              },
            ],
          ]
        }
      >
        Wilderness Camping
      </Text>
      <View
        accessibilityLabel="select"
        selected={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "black",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 28,
              "justifyContent": "center",
              "width": 28,
            },
          ]
        }
      >
        <Icon
          color="white"
          icon="check"
          onPress={[Function]}
          size={26}
        />
      </View>
    </View>
  </View>,
]
`;
